From e0596965c4d986f17ef969aa54a3160e89254aab Mon Sep 17 00:00:00 2001
From: Sergey Mitrofanov <sergeym@etersoft.ru>
Date: Wed, 30 Jan 2013 15:28:50 +0400
Subject: [PATCH 06/22] Now all paths write to config after start.
To: wine-patches <wine-patches@winehq.org>
Reply-To: wine-devel <wine-devel@winehq.org>

Conflicts:
	opennx/opennxApp.cpp
---
 opennx/SessionProperties.cpp | 19 +++++-----
 opennx/opennxApp.cpp         | 29 ++++++++-------
 opennx/res/opennx.xrc        | 85 --------------------------------------------
 3 files changed, 26 insertions(+), 107 deletions(-)

diff --git a/opennx/SessionProperties.cpp b/opennx/SessionProperties.cpp
index 0587e97..564b5c4 100644
--- a/opennx/SessionProperties.cpp
+++ b/opennx/SessionProperties.cpp
@@ -1012,9 +1012,10 @@ void SessionProperties::CreateControls()
     m_pCtrlUsbAdd = XRCCTRL(*this, "ID_BUTTON_USBADD", wxButton);
     m_pCtrlUsbModify = XRCCTRL(*this, "ID_BUTTON_USBMODIFY", wxButton);
     m_pCtrlUsbDelete = XRCCTRL(*this, "ID_BUTTON_USBDELETE", wxButton);
-    m_pCtrlUserNxDir = XRCCTRL(*this, "ID_TEXTCTRL_USERDIR", wxTextCtrl);
-    m_pCtrlSystemNxDir = XRCCTRL(*this, "ID_TEXTCTRL_SYSDIR", wxTextCtrl);
-    m_pCtrlCupsPath = XRCCTRL(*this, "ID_TEXTCTRL_CUPSPATH", wxTextCtrl);
+    //sergeym
+    //m_pCtrlUserNxDir = XRCCTRL(*this, "ID_TEXTCTRL_USERDIR", wxTextCtrl);
+    //m_pCtrlSystemNxDir = XRCCTRL(*this, "ID_TEXTCTRL_SYSDIR", wxTextCtrl);
+    //m_pCtrlCupsPath = XRCCTRL(*this, "ID_TEXTCTRL_CUPSPATH", wxTextCtrl);
     m_pCtrlCupsBrowse = XRCCTRL(*this, "ID_BUTTON_BROWSE_CUPSPATH", wxButton);
     m_pCtrlUsbipdDaemon = XRCCTRL(*this, "ID_PANEL_USBIP_DAEMON", wxPanel);
     m_pCtrlUsbIpdSocket = XRCCTRL(*this, "ID_TEXTCTRL_USBIPD_SOCKET", wxTextCtrl);
@@ -1087,14 +1088,14 @@ void SessionProperties::CreateControls()
         FindWindow(XRCID("ID_CHECKBOX_MMEDIA"))->SetValidator( wxGenericValidator(& m_bEnableMultimedia) );
     if (FindWindow(XRCID("ID_CHECKBOX_USBENABLE")))
         FindWindow(XRCID("ID_CHECKBOX_USBENABLE"))->SetValidator( wxGenericValidator(& m_bEnableUSBIP) );
-    if (FindWindow(XRCID("ID_TEXTCTRL_USERDIR")))
-        FindWindow(XRCID("ID_TEXTCTRL_USERDIR"))->SetValidator( MyValidator(& m_sUserNxDir) );
+//    if (FindWindow(XRCID("ID_TEXTCTRL_USERDIR")))
+  //      FindWindow(XRCID("ID_TEXTCTRL_USERDIR"))->SetValidator( MyValidator(& m_sUserNxDir) );
     if (FindWindow(XRCID("ID_CHECKBOX_REMOVEOLDSF")))
         FindWindow(XRCID("ID_CHECKBOX_REMOVEOLDSF"))->SetValidator( wxGenericValidator(& m_bRemoveOldSessionFiles) );
-    if (FindWindow(XRCID("ID_TEXTCTRL_SYSDIR")))
-        FindWindow(XRCID("ID_TEXTCTRL_SYSDIR"))->SetValidator( MyValidator(& m_sSystemNxDir) );
-    if (FindWindow(XRCID("ID_TEXTCTRL_CUPSPATH")))
-        FindWindow(XRCID("ID_TEXTCTRL_CUPSPATH"))->SetValidator( MyValidator(& m_sCupsPath) );
+//    if (FindWindow(XRCID("ID_TEXTCTRL_SYSDIR")))
+  //      FindWindow(XRCID("ID_TEXTCTRL_SYSDIR"))->SetValidator( MyValidator(& m_sSystemNxDir) );
+ //   if (FindWindow(XRCID("ID_TEXTCTRL_CUPSPATH")))
+   //     FindWindow(XRCID("ID_TEXTCTRL_CUPSPATH"))->SetValidator( MyValidator(& m_sCupsPath) );
     if (FindWindow(XRCID("ID_TEXTCTRL_USBIPD_SOCKET")))
         FindWindow(XRCID("ID_TEXTCTRL_USBIPD_SOCKET"))->SetValidator( MyValidator(& m_sUsbipdSocket) );
     if (FindWindow(XRCID("ID_SPINCTRL_USB_LOCALPORT")))
diff --git a/opennx/opennxApp.cpp b/opennx/opennxApp.cpp
index 0f043c4..3a5f83e 100644
--- a/opennx/opennxApp.cpp
+++ b/opennx/opennxApp.cpp
@@ -400,8 +400,9 @@ opennxApp::CheckDesktopEntry(MyXmlConfig *cfg)
 opennxApp::setUserDir()
 {
     wxString tmp;
-    if (!wxConfigBase::Get()->Read(wxT("Config/UserNxDir"), &tmp))
-        tmp = ::wxGetHomeDir() + wxFileName::GetPathSeparator() + wxT(".nx");
+    //sergeym
+    //if (!wxConfigBase::Get()->Read(wxT("Config/UserNxDir"), &tmp))
+    tmp = ::wxGetHomeDir() + wxFileName::GetPathSeparator() + wxT(".nx");
     wxFileName::Mkdir(tmp, 0750, wxPATH_MKDIR_FULL);
     wxFileName fn(tmp);
     wxConfigBase::Get()->Write(wxT("Config/UserNxDir"), fn.GetFullPath());
@@ -522,8 +523,9 @@ opennxApp::preInit()
     wxString appver;
     wxString thisver(wxT(PACKAGE_VERSION));
     thisver.Append(wxT(".")).Append(wxT(SVNREV));
+    //sergeym
     wxConfigBase::Get()->Read(wxT("Config/CurrentVersion"), &appver, thisver);
-    if (!thisver.IsSameAs(appver)) {
+    //if (!thisver.IsSameAs(appver)) {
         wxFileName fn(GetSelfPath());
         if (fn.GetDirs().Last().IsSameAs(wxT("bin")))
             fn.RemoveLastDir();
@@ -534,7 +536,9 @@ opennxApp::preInit()
         if (thissysdir.EndsWith(sep, &rest))
             thissysdir = rest;
         wxString cfgsysdir;
-        if (wxConfigBase::Get()->Read(wxT("Config/SystemNxDir"), &cfgsysdir)) {
+        wxConfigBase::Get()->Write(wxT("Config/SystemNxDir"), thissysdir);
+        wxConfigBase::Get()->Flush();
+       /* if (wxConfigBase::Get()->Read(wxT("Config/SystemNxDir"), &cfgsysdir)) {
             if (!thissysdir.IsSameAs(cfgsysdir)) {
                 if (!m_bNoGui) {
                     wxString msg(wxString::Format(_("Your System NX directory setting (%s)\nappears to be incorrect.\nDo you want to change it to the correct default value\n(%s)?"), cfgsysdir.c_str(), thissysdir.c_str()));
@@ -544,9 +548,9 @@ opennxApp::preInit()
                         wxConfigBase::Get()->Flush();
                     }
                 }
-            }
-        }
-    }
+            //}
+        }*/
+    //}
     wxConfigBase::Get()->Write(wxT("Config/CurrentVersion"), thisver);
     wxConfigBase::Get()->Flush();
 
@@ -570,7 +574,8 @@ opennxApp::preInit()
     m_cLocale.Init();
     m_cLocale.AddCatalog(wxT("opennx"));
 
-    if (!wxConfigBase::Get()->Read(wxT("Config/CupsPath"), &tmp)) {
+   //sergeym
+  //  if (!wxConfigBase::Get()->Read(wxT("Config/CupsPath"), &tmp)) {
 #if defined(__LINUX__) || defined(__OPENBSD__) || defined(__WXMAC__)
         tmp = findExecutable(wxT("cupsd"));
         if (tmp.IsEmpty()) {
@@ -599,11 +604,9 @@ opennxApp::preInit()
             }
         }
 #endif
-        if (!tmp.IsEmpty()) {
-            wxConfigBase::Get()->Write(wxT("Config/CupsPath"), tmp);
-            wxConfigBase::Get()->Flush();
-        }
-    }
+        wxConfigBase::Get()->Write(wxT("Config/CupsPath"), tmp);
+        wxConfigBase::Get()->Flush();
+  //  }
 #ifdef SUPPORT_USBIP
     if (!wxConfigBase::Get()->Read(wxT("Config/UsbipdSocket"), &tmp)) {
         tmp = wxT("/var/run/usbipd2.socket");
diff --git a/opennx/res/opennx.xrc b/opennx/res/opennx.xrc
index 27936ac..5697a0e 100644
--- a/opennx/res/opennx.xrc
+++ b/opennx/res/opennx.xrc
@@ -1014,28 +1014,6 @@
                                                 <orient>wxVERTICAL</orient>
                                                 <label>User NX directory</label>
                                                 <object class="sizeritem">
-                                                    <flag>wxGROW</flag>
-                                                    <border>5</border>
-                                                    <object class="wxBoxSizer">
-                                                        <orient>wxHORIZONTAL</orient>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <option>1</option>
-                                                            <object class="wxTextCtrl" name="ID_TEXTCTRL_USERDIR">
-                                                            </object>
-                                                        </object>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <object class="wxButton" name="ID_BUTTON_BROWSE_USERDIR">
-                                                                <style>wxBU_EXACTFIT</style>
-                                                                <label>...</label>
-                                                            </object>
-                                                        </object>
-                                                    </object>
-                                                </object>
-                                                <object class="sizeritem">
                                                     <flag>wxGROW|wxALL</flag>
                                                     <border>5</border>
                                                     <object class="wxCheckBox" name="ID_CHECKBOX_REMOVEOLDSF">
@@ -1046,69 +1024,6 @@
                                             </object>
                                         </object>
                                         <object class="sizeritem">
-                                            <flag>wxGROW|wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                            <border>5</border>
-                                            <object class="wxStaticBoxSizer" name="wxID_ANY">
-                                                <orient>wxVERTICAL</orient>
-                                                <label>System NX directory</label>
-                                                <object class="sizeritem">
-                                                    <flag>wxGROW</flag>
-                                                    <border>5</border>
-                                                    <option>1</option>
-                                                    <object class="wxBoxSizer">
-                                                        <orient>wxHORIZONTAL</orient>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <option>1</option>
-                                                            <object class="wxTextCtrl" name="ID_TEXTCTRL_SYSDIR">
-                                                            </object>
-                                                        </object>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <object class="wxButton" name="ID_BUTTON_BROWSE_SYSDIR">
-                                                                <style>wxBU_EXACTFIT</style>
-                                                                <label>...</label>
-                                                            </object>
-                                                        </object>
-                                                    </object>
-                                                </object>
-                                            </object>
-                                        </object>
-                                        <object class="sizeritem">
-                                            <flag>wxGROW|wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                            <border>5</border>
-                                            <option>1</option>
-                                            <object class="wxStaticBoxSizer" name="wxID_ANY">
-                                                <orient>wxVERTICAL</orient>
-                                                <label>System CUPS daemon</label>
-                                                <object class="sizeritem">
-                                                    <flag>wxGROW</flag>
-                                                    <border>5</border>
-                                                    <option>1</option>
-                                                    <object class="wxBoxSizer">
-                                                        <orient>wxHORIZONTAL</orient>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <option>1</option>
-                                                            <object class="wxTextCtrl" name="ID_TEXTCTRL_CUPSPATH">
-                                                            </object>
-                                                        </object>
-                                                        <object class="sizeritem">
-                                                            <flag>wxALIGN_CENTER_VERTICAL|wxALL</flag>
-                                                            <border>5</border>
-                                                            <object class="wxButton" name="ID_BUTTON_BROWSE_CUPSPATH">
-                                                                <style>wxBU_EXACTFIT</style>
-                                                                <label>...</label>
-                                                            </object>
-                                                        </object>
-                                                    </object>
-                                                </object>
-                                            </object>
-                                        </object>
-                                        <object class="sizeritem">
                                             <flag>wxGROW|wxALIGN_CENTER_VERTICAL|wxLEFT|wxRIGHT</flag>
                                             <border>5</border>
                                             <object class="wxPanel" name="ID_PANEL_USBIP_DAEMON">
-- 
2.1.4

